certification/internal/shell/has_prohibited_packages_mounted.go

```diff
@@ -0,0 +1,113 @@
+package shell
+
+import (
+	"path/filepath"
+	"strings"
+
+	rpmdb "github.com/knqyf263/go-rpmdb/pkg"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+type HasNoProhibitedPackagesMountedCheck struct{}
+
+func (p *HasNoProhibitedPackagesMountedCheck) Validate(image string) (bool, error) {
+	pkgList, err := p.getDataToValidate(image)
+	if err != nil {
+		log.Error("unable to get a list of all packages in the image")
+		return false, err
+	}
+
+	return p.validate(pkgList)
+}
+
+func (p *HasNoProhibitedPackagesMountedCheck) getDataToValidate(dir string) ([]string, error) {
+	log.Debugf("Mounted directory: %s", dir)
+
+	db, err := rpmdb.Open(filepath.Join(dir, "var", "lib", "rpm", "Packages"))
+	if err != nil {
+		return nil, err
+	}
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return nil, err
+	}
+	pkgs := make([]string, len(pkgList))
+	for i, pkg := range pkgList {
+		pkgs[i] = pkg.Name
+	}
+	return pkgs, nil
+}
+
+func (p *HasNoProhibitedPackagesMountedCheck) validate(pkgList []string) (bool, error) {
+	var prohibitedPackages []string
+	for _, pkg := range pkgList {
+		_, ok := prohibitedPackageList[pkg]
+		if ok {
+			prohibitedPackages = append(prohibitedPackages, pkg)
+			continue
+		}
+		for _, name := range prohibitedPackageGlobList {
+			if strings.HasPrefix(pkg, name) {
+				prohibitedPackages = append(prohibitedPackages, pkg)
+				continue
+			}
+		}
+	}
+
+	log.Warn("The number of prohibited package found in the container image: ", len(prohibitedPackages))
+	if len(prohibitedPackages) > 0 {
+		log.Warn("found the following prohibited packages: ", prohibitedPackages)
+	}
+
+	return len(prohibitedPackages) == 0, nil
```

Comment:
nit: Can you instead evaluate `len(prohibitedPackages) == 0` and return `true, nil`, and otherwise warn as you've done on 60 outside of the loop, and return `false, nil`? I think on the whole that I prefer the explicit returns over the internally evaluated cases such as line line 63.

Reply:
Sure